internal/dao/sqlx/reservation_dao.go

```diff
@@ -204,11 +241,26 @@ func (di *reservationDaoSqlx) UpdateReservationIDForAWS(ctx context.Context, id
 	return nil
 }
 
-func (di *reservationDaoSqlx) Finish(ctx context.Context, id int64, success bool, status string) error {
-	query := finishReservationStatus
-	stmt := di.finishStatus
+func (di *reservationDaoSqlx) FinishWithSuccess(ctx context.Context, id int64) error {
+	query := finishReservationSuccess
+	stmt := di.finishSuccess
+
+	res, err := stmt.ExecContext(ctx, id, true)
+	if err != nil {
+		return NewExecUpdateError(ctx, di, query, err)
+	}
+	if rows, _ := res.RowsAffected(); rows != 1 {
+		return NewUpdateMismatchAffectedError(ctx, di, 1, rows)
+
+	}
+	return nil
+}
+
+func (di *reservationDaoSqlx) FinishWithError(ctx context.Context, id int64, errorString string) error {
+	query := finishReservationError
+	stmt := di.finishError
 
-	res, err := stmt.ExecContext(ctx, id, status, success)
+	res, err := stmt.ExecContext(ctx, id, false, errorString)
 	if err != nil {
 		return NewExecUpdateError(ctx, di, query, err)
 	}
```

Comment:
These two finish functions almost do the same thing, yet the latter takes `error` field not `status` field.

I really do not want to create a generic `Update(ctx, reservation)` function because that will make the handler code really not very readable.